src/runners/utils.ts

```diff
@@ -0,0 +1,79 @@
+import * as bodyParser from 'body-parser';
+import * as cors from 'cors';
+import * as express from 'express';
+// tslint:disable-next-line:no-implicit-dependencies
+import * as core from 'express-serve-static-core';
+import { createServer, Server } from 'http';
+
+import { AppDependencies } from '../app';
+import { HEALTHCHECK_PATH } from '../constants';
+import { logger } from '../logger';
+import { requestLogger } from '../middleware/request_logger';
+import { createHealthcheckRouter } from '../routers/healthcheck_router';
+import { HealthcheckService } from '../services/healthcheck_service';
+import { HttpServiceConfig } from '../types';
+
+/**
+ * creates the NodeJS http server with graceful shutdowns, healthchecks,
+ * configured header timeouts and other sane defaults set.
+ */
+export function createDefaultServer(
+    dependencies: AppDependencies,
+    config: HttpServiceConfig,
+    app: core.Express,
+): Server {
+    app.use(requestLogger());
+    app.use(cors());
+    app.use(bodyParser.json());
+
+    const server = createServer(app);
+    server.keepAliveTimeout = config.httpKeepAliveTimeout;
+    server.headersTimeout = config.httpHeadersTimeout;
+    const healthcheckService = new HealthcheckService();
+
+    server.on('close', () => {
+        logger.info('http server shutdown');
+    });
+    server.on('listening', () => {
+        logger.info(`server listening on ${config.httpPort}`);
+        healthcheckService.setHealth(true);
+    });
+
+    const shutdownFunc = (sig: string) => {
+        logger.info(`received: ${sig}, shutting down server`);
+        healthcheckService.setHealth(false);
+        server.close(async err => {
+            if (!server.listening) {
+                process.exit(0);
+            }
+            if (dependencies.connection) {
+                await dependencies.connection.close();
+            }
+            if (dependencies.meshClient) {
+                dependencies.meshClient.destroy();
```

Comment:
just wanna confirm this is **not** async, right?

Reply:
Correct, it is not async :)